### PR TITLE
test and doc IR Graph

### PR DESCRIPTION
**Pull request by panyx0718**

### CODE REVIEW

File: paddle/fluid/framework/ir/graph_test.cc [hunk 1]
```diff
@@ -110,5 +126,83 @@ TEST(GraphTest, Basic) {
...
+  op->SetType("sum");
+  op->SetInput("X", {"a"});
+  op->SetOutput("Out", {"b"});
+  op->SetAttr("op_role", 1);
+
+  op = prog.MutableBlock(0)->AppendOp();
+  op->SetType("dummy");
+  op->SetInput("X", {"c"});
+  op->SetOutput("Out", {"a"});
+  op->SetAttr("op_role", 1);
```
Comment (chengduoZH) [inline paddle/fluid/framework/ir/graph_test.cc:143]: Do you want to express `a_var->sum_op->b_var->dummy_op->a_var`?

**Review by Superjomn — APPROVED** [review]
LGTM

File: paddle/fluid/framework/ir/graph_test.cc (hunk 1 above)
Comment (panyx0718) [reply paddle/fluid/framework/ir/graph_test.cc:143]: no, dummy_op's input is c. This is used to test WAR of variable a.

File: paddle/fluid/framework/ir/graph_test.cc [hunk 2]
```diff
@@ -110,5 +126,83 @@ TEST(GraphTest, Basic) {
...
+  prog.MutableBlock(0)->Var("b")->SetType(proto::VarType::LOD_TENSOR);
+  prog.MutableBlock(0)->Var("c")->SetType(proto::VarType::LOD_TENSOR);
+
+  std::unique_ptr<ir::Graph> g(new ir::Graph(prog));
+  ir::Node *control_dep1 = nullptr;
+  ir::Node *control_dep2 = nullptr;
+  for (ir::Node *n : g->Nodes()) {
+    if (n->Name() == "sum") {
+      ASSERT_EQ(n->outputs[0]->Name(), "b");
+      ASSERT_TRUE(ir::IsControlDepVar(*n->outputs[1]));
```
Comment (dzhwinter) [inline paddle/fluid/framework/ir/graph_test.cc:155]: a little concern about the test. Here we assume sum op, dummy op has multiple outputs, one of them is dep_var. However, in our analysis process, we do not have additional outputs slot for the dep_var, how will we handle the WAR, WAW harzard?

**Pull request merged**